### PR TITLE
ℹ️ About & Help Screen

### LINKED ISSUE #526: Settings: Implement Help
## Description
Same as the old app.

[iOS Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19800511) [Android Designs page.](https://toggl.invisionapp.com/d/main?origin=v7#/projects/prototypes/19932960) [Notion page.](https://www.notion.so/toggl/Settings)

## Dependencies
`Settings: Add settings view` #512
## Definition of done
- [ ] Implement the logic
- [ ] implement the UI

### PR DESCRIPTION
**Pull request by semanticer**
### Summary
Content of this PR:

- Help screen
- About screen
- Privacy Policy screen
- Terms of Service screen
- Licenses
- Small Settings refactors
- Small feedback refactors

### Technical considerations
**Help**, **Privacy Policy** and **Terms of Service** simply launch Intent for browser. The Intent is fired as soon as we set `externalLocationToShow` property in the local settings state. 

**About screen** is heavily reusing stuff we have already in the main settings screen. I'm using the same view models **SubPage** and even the same selector where I simply inject a different blueprint. 

 **Licenses screen** is implemented with the help of [google's gradle plugin](https://developers.google.com/android/guides/opensource) the OssLicensesMenuActivity is launched using our standard navigation patterns.

**Small Settings refactors** I've made settings ui package even more **package by feature** I think it makes a lot of sense.

**Small feedback refactors** I've just made it more consistent, I'm now using dispatcher in composable functions. 

### Why is it done like this and not some other way?
The intention was to make this easy as possible with the help of some code-reuse. I also made sure it will be easy to change settings blueprints and move stuff around without major refactors. 

### Relationships

Closes #526 
Closes #627 

### Pull Request checklist

- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Should be fairly easy to review everything at once.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Anyone

### CODE REVIEW

File: settings/src/main/java/com/toggl/settings/ui/SettingsFragment.kt [hunk 1]
```diff
@@ -48,7 +58,26 @@ class SettingsFragment : Fragment() {
...
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        store.state
+            .map { it.localState.externalLocationToShow }
+            .filterNotNull()
+            .distinctUntilChanged()
+            .map { it.getExternalUri(requireContext()) }
+            .onEach { externalLocationUri ->
+                val openURL = Intent(Intent.ACTION_VIEW)
```
Comment (daividssilverio) [inline settings/src/main/java/com/toggl/settings/ui/SettingsFragment.kt:68]: Ideally, we would use Chrome custom tabs if they are available. That's how it works on the old app.

File: settings/src/main/java/com/toggl/settings/ui/common/ToggleSetting.kt [hunk 2]
```diff
@@ -39,7 +39,10 @@ fun PreviewToggleLight() {
     ThemedPreview {
         Column {
             SettingsRow(previewToggle) { }
-            SettingsRow(previewToggle.copy(toggled = true)) { }
+            SettingsRow(
+                    previewToggle.copy(
+                        toggled = true
+                    )) { }
```
[the remark this answers is not in the document]
Comment (semanticer) [reply settings/src/main/java/com/toggl/settings/ui/common/ToggleSetting.kt:45]: AS auto-formatting did this!

File: settings/src/main/java/com/toggl/settings/ui/common/Section.kt [hunk 3]
```diff
@@ -82,9 +84,9 @@ internal fun SettingsRow(
     val rowModifier = Modifier
         .height(48.dp)
         .fillMaxWidth() + Modifier.clickable(
-            indication = RippleIndication(),
-            onClick = onClickAction
-        )
+        indication = RippleIndication(),
+        onClick = onClickAction
+    )
```
[the remark this answers is not in the document]
Comment (semanticer) [reply settings/src/main/java/com/toggl/settings/ui/common/Section.kt:89]: wtf happened to my formating? 🤔

File: common-android/src/main/res/values/external_links.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="help_link" translatable="false">"https://support.toggl.com/en/articles/2417167-toggl-timer-for-android"</string>
+    <string name="privacy_policy_link" translatable="false">"https://toggl.com/legal/privacy/"</string>
+    <string name="terms_of_service_link" translatable="false">"https://toggl.com/legal/terms/"</string>
```
[the remark this answers is not in the document]
Comment (semanticer) [reply common-android/src/main/res/values/external_links.xml:5]: I've found the thread you started https://toggl.slack.com/archives/C0170MEUGJV/p1594642407005800 looks like there is no consensus yet, I will investigate more.

[the remark this answers is not in the document]
Comment (heytherewill) [reply common-android/src/main/res/values/external_links.xml:5]: Feel free to just create a pumbling issue so we don't forget, but it shouldn't block this PR

**Comment by semanticer** [comment]
@heytherewill 
> in for a penny in for a pound I guess

Indeed
I 🚧 everything so that we only have one SectionSelector and did a bunch of other stuff to streamline settings screens. Let me know what you think. 

@daividssilverio 

> The external webpages are own content... why aren't we using chrome custom tabs again? I guess this was discussed somewhere, but it just doesn't look nice.

Good point, I didn't know we're using chrome tabs before. 
It's implemented now, without a custom toolbar color for now... let's wait for designers to choose the right color.

> The licenses activity is lacking a title, it looks broken. There's a OssLicensesMenuActivity.setActivityTitle, but it seems to be broken :/ Maybe we could add an extra with the activity title when navigating, not sure how to do that though.

Let's skip this, for now, I'll create an issue and maybe they will fix the title setter before we release Aurora :-D 

> Can't we make the dispatchers for Section, SettingsRow optional? We are adding stuff like ...

I would prefer not to do so, we are using empty lambdas only in preview code and we still want to 'force' clients to pass dispatcher in non-preview code.

**Pull request merged**